Documentation/tls/rotate-tls.md

```diff
@@ -1,929 +1,338 @@
+$ kubectl --kubeconfig generated/auth/kubeconfig get nodes
+NAME                                        STATUS    ROLES     AGE       VERSION
+ip-10-0-44-164.us-west-1.compute.internal   Ready     node      14m       v1.8.4+coreos.0
+ip-10-0-45-138.us-west-1.compute.internal   Ready     node      14m       v1.8.4+coreos.0
+ip-10-0-52-88.us-west-1.compute.internal    Ready     node      14m       v1.8.4+coreos.0
+ip-10-0-6-59.us-west-1.compute.internal     Ready     master    14m       v1.8.4+coreos.0
 ```
 
-Tectonic Console should now use the new certificate for subsequent
-operations. Log out and back into Tectonic Console to verify everything is working properly.
+Errors encountered during this step indicate that one or more of the input values was incorrect.
 
-## etcd
+## Rotating certificates for Tectonic and Kubernetes components.
 
-There are several TLS certificates to be managed when administering an etcd
-cluster. This example will generate a new CA certificate and use this CA to generate new self-signed certificates for each of the components
-in the etcd cluster.
+TLS assets for components running on top of Kubernetes can be updated using `kubectl`, including self-hosted control plane components such as the API server. To rotate those certificates, patch the manifests and roll the deployments.
 
-### Verify cluster health
-
-First, verify the current health of the etcd cluster. Connect to one of the etcd members of the cluster using SSH.
+__WARNING:__ The following commands MUST use `kubectl patch` and NOT other `kubectl` creation sub-commands.
 
 ```
-ssh core@<ETCD MEMBER NODE>
+kubectl --kubeconfig generated/auth/kubeconfig \
+    patch -f generated/patches/identity-grpc-client.patch \
+    -p "$( cat generated/patches/identity-grpc-client.patch )"
+kubectl --kubeconfig generated/auth/kubeconfig \
+    patch -f generated/patches/identity-grpc-server.patch \
+    -p "$( cat generated/patches/identity-grpc-server.patch )"
+kubectl --kubeconfig generated/auth/kubeconfig \
+    patch -f generated/patches/ingress-tls.patch \
+    -p "$( cat generated/patches/ingress-tls.patch )"
+kubectl --kubeconfig generated/auth/kubeconfig \
+    patch -f generated/patches/kube-apiserver-secret.patch \
+    -p "$( cat generated/patches/kube-apiserver-secret.patch )"
 ```
 
-etcd clusters should be configured to require client authentication. Therefore, we will need the existing CA certificate, and the client certificate and key for the cluster. These artifacts should be located in the `/etc/ssl/etcd` directory if the Tectonic cluster was set up using self-signed certificates.
+To force the various deployments to restart and pick up the new TLS assets, force the rotation of the various components. Note that the API server may become temporarily unavailable after this action.
```

Comment:
To force the deployments to restart and pick up the new TLS assets, force the rotation of the deployments' components.